src/app/shared/deactivate-device/vault.service.ts

```diff
@@ -0,0 +1,37 @@
+/*********************************************************************
+* Copyright (c) Intel Corporation 2021
+* SPDX-License-Identifier: Apache-2.0
+**********************************************************************/
+
+import { HttpClient } from '@angular/common/http'
+import { Injectable } from '@angular/core'
+import { Observable, throwError } from 'rxjs'
+import { catchError } from 'rxjs/operators'
+import { AuthService } from 'src/app/auth.service'
+import { environment } from 'src/environments/environment'
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class VaultService {
+  server: string = environment.rpsServer.replace('/rps', ':8200').replace('https', 'http')
```

Comment:
this won't work if not using kong. I think a better way would be to substring after `://` and up to `:` or `/`

Reply:
Instead of using rpsServer, added new variable in environment for vault